ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -92,162 +91,210 @@ public void run(DatabaseClient client, InputStream csvRecords, InputStream jobCo
-               .withBatchSize(getBatchSize());
-			if (logger != null) {
+		final JobRunnerVariables jobRunnerVariables = new JobRunnerVariables();
+
+		try {
+			ObjectMapper objectMapper = new ObjectMapper();
+
+			ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobControl);
+			ObjectNode jobDef = (ObjectNode) jobControlObj.path("job");
+			ObjectNode recordDef = (ObjectNode) jobControlObj.path("record");
+			if (jobDef.isMissingNode() || recordDef.isMissingNode()) {
+				throw new MarkLogicIOException("job Node and/or record Node(s) cannot be empty.");
+			}
+			String id = jobDef.path("id").asText();
+			if (id == null || id.length() == 0) {
+				throw new MarkLogicIOException("job id cannot be empty or Null");
+			}
+			this.jobId = id;
+
+			JsonNode metadataNode = jobDef.path("metadata");
+			if(metadataNode.isMissingNode()) {
+				throw new MarkLogicIOException("metadata node cannot be empty.");
+			}
+			DocumentMetadataHandle documentMetadata = new DocumentMetadataHandle();
+			documentMetadata.withCollections(getJobCollection(id));
+			DocumentPermissions permissions = documentMetadata.getPermissions();
+			for (String temp : getRoles()) {
+				permissions.add(temp, Capability.READ, Capability.UPDATE);
+			}
+
+			CSVConverter converter = new CSVConverter();
+			Iterator<ObjectNode> itr = converter.convertObject(csvRecords);
+			if (!itr.hasNext()) {
+				throw new MarkLogicIOException("No header found.");
+			}
+
+			WriteBatcher batcher = moveMgr.newWriteBatcher().withBatchSize(getBatchSize());
+			if (jobLogger != null) {
 				batcher = batcher.onBatchSuccess(batch -> {
-						long lastTime = jobRunnerVariables.lastSuccessLogTime.longValue();
-						long currentTime = System.currentTimeMillis();
-						long threshhold = lastTime + (getTimeIntervalInSeconds() * 1000);
-						long currentTotalSuccessCount = jobRunnerVariables.totalSuccessCount.incrementAndGet();
-						if (threshhold <= currentTime
-								&& jobRunnerVariables.lastSuccessLogTime.compareAndSet(lastTime, currentTime)) {
-							logger.info("Success for batch number - " + batch.getJobBatchNumber()
-									+ ". Batch running since last log - "
-									+ (currentTotalSuccessCount - jobRunnerVariables.lastSuccessCount.get())
-									+ ". Total successes - " + currentTotalSuccessCount);
-							jobRunnerVariables.lastSuccessCount.set(currentTotalSuccessCount);
+
+					long lastTime = jobRunnerVariables.lastSuccessLogTime.longValue();
+					long currentTime = System.currentTimeMillis();
+					long threshhold = lastTime + (getTimeIntervalInSeconds() * 1000);
+					long currentTotalSuccessCount = jobRunnerVariables.totalSuccessCount.incrementAndGet();
+					if (threshhold <= currentTime
+							&& jobRunnerVariables.lastSuccessLogTime.compareAndSet(lastTime, currentTime)) {
+						ObjectNode statusNode = new ObjectMapper().createObjectNode();
+						statusNode.put("Timestamp", new Timestamp(System.currentTimeMillis()).toString());
```

Comment:
Could the status object have a "Current status" property similar to all of the other status messages?